packages/terra-folder-tree/src/subcomponents/FolderTreeItem.jsx

```diff
@@ -112,6 +123,48 @@ const FolderTreeItem = ({
     }
   };
 
+  const handleKeyDown = event => {
+    switch (event.nativeEvent.keyCode) {
+      case KeyCode.KEY_RETURN:
+      case KeyCode.KEY_SPACE:
+        event.preventDefault();
+
+        if (isFolder) {
+          handleToggle(event);
+        } else {
+          onClick(event);
```

Comment:
This isn't something we need to do right now, but I think `onClick` would be better renamed to `onSelect` to be more reflective of its actual usage.